database/src/episode.rs

```diff
@@ -218,89 +123,27 @@ impl Episode {
     /// * `conn` - diesel connection reference to postgres
     /// * `media` - reference to a season object/entry.
     ///
-    /// # Example
-    /// ```
-    /// use database::get_conn_devel as get_conn;
-    /// use database::library::{Library, InsertableLibrary, MediaType};
-    /// use database::media::{InsertableMedia, Media};
-    /// use database::tv::InsertableTVShow;
-    /// use database::season::{InsertableSeason, Season};
-    /// use database::episode::{InsertableEpisode, Episode};
-    ///
-    /// let conn = get_conn().unwrap();
-    ///
-    /// let library = InsertableLibrary {
-    ///     name: "test".into(),
-    ///     location: "/dev/null".to_string(),
-    ///     media_type: MediaType::Tv,
-    /// };
-    ///
-    /// let library_id = library.insert(&conn).unwrap();
-    ///
-    /// let new_show = InsertableMedia {
-    ///     library_id: library_id,
-    ///     name: "test".into(),
-    ///     added: "test".into(),
-    ///     media_type: MediaType::Tv,
-    ///     ..Default::default()
-    /// };
-    ///
-    /// let show_id = new_show.into_static::<InsertableTVShow>(&conn).unwrap();
-    ///
-    /// let new_season = InsertableSeason {
-    ///     season_number: 1,
-    ///     added: "test".into(),
-    ///     poster: "test".into(),
-    /// };
-    ///
-    /// let season_id = new_season.insert(&conn, show_id).unwrap();
-    ///
-    /// let new_episode = InsertableEpisode {
-    ///     media: InsertableMedia {
-    ///         library_id: library_id,
-    ///         name: "test_episode".into(),
-    ///         added: "test".into(),
-    ///         media_type: MediaType::Episode,
-    ///         ..Default::default()
-    ///     },
-    ///     seasonid: season_id,
-    ///     episode: 1,
-    /// };
-    ///
-    /// let episode_id = new_episode.insert(&conn, show_id).unwrap();
-    ///
-    /// let show = Media::get(&conn, show_id).unwrap();
-    /// let season = Season::get(&conn, show_id, 1).unwrap();
-    ///
-    /// let episodes = Episode::get_all_of_season(&conn, &season).unwrap();
-    ///
-    /// let episode = &episodes[0];
-    /// assert_eq!(episode.id, episode_id);
-    /// assert_eq!(episode.episode, 1);
-    ///
-    /// Library::delete(&conn, library_id).unwrap();
     pub async fn get_all_of_season(
         conn: &crate::DbConnection,
         media: &Season,
     ) -> Result<Vec<Episode>, DatabaseError> {
-        use crate::schema::media;
-
-        let wrappers = episode::dsl::episode
-            .filter(episode::seasonid.eq(media.id))
-            .load_async::<EpisodeWrapper>(conn)
-            .await?;
+        let wrappers = sqlx::query_as!(
```

Comment:
entire function could be collapsed into one query with two joins.

Reply:
deferred